ledger/block/src/transaction/merkle.rs

```diff
@@ -116,18 +116,23 @@ impl<N: Network> Transaction<N> {
     pub fn deployment_tree(deployment: &Deployment<N>) -> Result<DeploymentTree<N>> {
         // Ensure the number of leaves is within the Merkle tree size.
         Self::check_deployment_size(deployment)?;
-        // Retrieve the program.
-        let program = deployment.program();
+        // Prepare the header for the hash.
+        let header = match deployment.program_checksum() {
```

Comment:
Attention: The rules for a deployment ID's preimage are changed here.

Now, we use the `program_checksum` instead of the `program_id` in the transaction leaf.

Reply:
See this commit: [0e274f1](https://github.com/ProvableHQ/snarkVM/pull/2655/commits/0e274f124a48fd47451d937c89fbe4c2aecc32ce)